mathics/core/symbols.py

```diff
@@ -410,7 +410,7 @@ def __new__(cls, name: str, sympy_dummy=None, value=None):
             # self.sympy_dummy, for which I have to dig into the
             # code to see even what type of value should be expected
             # for it.
-            self.value = value
+            self._value = value
```

Comment:
Except for some immutable symbols, it does not make sense to attach a value to a Symbol. Values of Symbols are always obtained from a Definitions object.

Reply:
There is confusion here. The symbol value is its name. It is not the value that the symbolic variable holds. That should be in the definitions object.

Reply:
So, it is the same than it's name?

Reply:
I just looked at the code, and this is confusing, and could use a bit of cleaning up, now that I understand things better. 

I started doing this, but I think it best to do this as a new single isolated PR. 

But let me try to explain first...

In short `value` is the same as its `name` (with _possibly_ one exception) and the two fields should have the same value in the future. 

In contrast to the comments in `Symbol.__init__()`  I believe this should work like Lisp's Symbol or Python's `system.intern()` .

And I think that the pattern used in the PR for numbers like Integer, Complex, and Real should be followed more closely in `Symbol.__init__()`.   Here, I think I was bootstrapping code from one onto the other. But in the revision, here are places I have improved things in my view. So that should be backported to  `Symbol.__init()`.

So now let's compare and contrast say ``Integer(0)`` from ``Symbol("System`N")``.  The part that is similar is that we want singletons based on a parameter for these. For Integer,  a parameter is 0. for Symbol, a parameter is ``System`N``. But for either of these, there should just be one object. 

For numbers, the value is constant. We never need to go through a Definitions object for that.  For Symbols, for most of them, values can change and so we have to go through a Definitions object. 

But there _are_ a couple of Symbols like True, False, List which we want to treat as constant - a user is not allowed to redefine these. And there is therefore no need to consult Definitions for these in order to get their values. 

So a hack I added was for this add True and False was to add `value` parameter which served in place of the name ``System`True``. 

So this should get cleaned up and a lot of comments and docstrings in the existing code simplified.  As I said, I think I understand better what's up here, based on @mmatera 's remark that Definition entries acted like bindings in other interpreters.

It might be that PredefinedSymbol's init looks more like Integer and that value parameter removed from `Symbol`'s init and `name` and `value` made identical. 

We'll see.